### PR TITLE
Jerry update

### PR DESCRIPTION
**Pull request by LaszloLango**
IoT.js-DCO-1.0-Signed-off-by: László Langó llango.u-szeged@partner.samsung.com

### CODE REVIEW

**Comment by seanshpark** [comment]
> 5,630 src/native_js.h

Is it OK to add this file?

**Comment by LaszloLango** [comment]
@seanshpark, no. Accidentally added. I've updated the PR.

**Comment by seanshpark** [comment]
can you check with arm-linux (raspberry pi 2) build?

**Comment by LaszloLango** [comment]
It builds, but bails out when I try to run. I have to investigate what the problem is.

**Comment by seanshpark** [comment]
ok, thank you.

**Comment by akiss77** [comment]
I'm all in for the bump, with two requests:
- Could you bump jerry even further? I'd go with current jerry master.
- build.py needy a copyright year bump.

**Comment by LaszloLango** [comment]
@akiss77, sure. The current master would be better, but at first we must find out why IoT.js fails on RPi2 after the update.

**Comment by LaszloLango** [comment]
@akiss77, @seanshpark, I've updated the PR. Please check. It worked on RPi2.

I built with the following command:
`tools/build.py --clean --target-arch=arm --target-board=rpi2 --compile-flag="-static" --link-flag="-static"`

File: tools/build.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env python
 
-# Copyright 2015 Samsung Electronics Co., Ltd.
+# Copyright 2016 Samsung Electronics Co., Ltd.
```
Comment (akiss77) [inline tools/build.py]: This should be 2015-2016

File: deps/jerry [hunk 2]
```diff
@@ -1 +1 @@
-Subproject commit cc23c225ea7c03bfb0fd4f58352ab517ba26dc9e
+Subproject commit c7d33e98871187ad4777defc928233702c96b5bf
```
Comment (akiss77) [inline deps/jerry]: There are newer commits in master again :)

File: src/iotjs_binding.cpp [hunk 3]
```diff
@@ -256,7 +256,6 @@ JObject JObject::GetProperty(const char* name) {
       _obj_val.u.v_object,
       reinterpret_cast<const jerry_api_char_t*>(name),
       &res);
-  IOTJS_ASSERT(is_ok);
```
Comment (akiss77) [inline src/iotjs_binding.cpp]: I've run into this earlier but couldn't track down why this assert was here and why it's not needed anymore. Actually, it's a blocker now. Any idea why's that?

Comment (LaszloLango) [inline src/iotjs_binding.cpp]: This caused me a headache too. There where a bug in the Jerry API: https://github.com/Samsung/jerryscript/pull/1042

Currently the assert fails on `iotjs_module_tcp.cpp:86`

`JObject::GetProperty` returns with the result of the `jerry_api_get_object_field_value` call. Every GetProperty() checks the return value in runtime, so they are expecting that the property might not exist. There is no description for `JObject::GetProperty`. If we leave the assert there, then you can call the `GetProperty` with existing property names. IMHO, this is a too strict limit.

Comment (akiss77) [inline src/iotjs_binding.cpp]: I agree. Thanks for the details.

**Comment by LaszloLango** [comment]
I've updated to the latest Jerry (9f29cc1) and tested on RPi2.

File: cmake/config/arm-linux.cmake [hunk 4]
```diff
@@ -15,7 +15,7 @@
 include(CMakeForceCompiler)
 
 set(CMAKE_SYSTEM_NAME Linux)
-set(CMAKE_SYSTEM_PROCESSOR arm)
+set(CMAKE_SYSTEM_PROCESSOR armv7l-hf)
```
Comment (akiss77) [inline cmake/config/arm-linux.cmake]: Sorry for the late hit, I didn't realize before that this file also needs a copyright year update.

**Comment by akiss77** [comment]
After the file header update, LGTM

**Comment by LaszloLango** [comment]
@seanshpark, please review.

**Comment by akiss77** [comment]
(Note: my suggestion is that after the second LGTM arrives, you do a bump of the jerry hash again. Not to get stuck in an infinite loop, you have my LGTM for that in advance, as long as only the hash changes.)

**Comment by seanshpark** [comment]
LGTM, push

**Pull request merged**